### PR TITLE
Isdb 286 c100 child info

### PR DESCRIPTION
**Pull request by saravanakumar-palanichamy**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by adetolacog — COMMENTED** [review]
Could you change the test method name to reflect the form.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100ServiceTest.java [hunk 1]
```diff
@@ -42,114 +56,148 @@ class BulkScanC100ServiceTest {
...
+            .forEach(field -> field.setValue("false"));
+        bulkScanTransformationRequest.getOcrdatafields().stream()
+            .filter(eachField ->
+                        CHILD_LIVING_WITH_OTHERS.equalsIgnoreCase(eachField.getName()))
+            .forEach(field -> field.setValue("true"));
+        BulkScanTransformationResponse res = bulkScanValidationService.transform(bulkScanTransformationRequest);
+        assertEquals(
+            "OtherPeople",
+            res.getCaseCreationDetails().getCaseData().get(CHILD_LIVE_WITH_KEY));
     }
```
Comment (adetolacog) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100ServiceTest.java:202]: Should testC100ValidationErrorWithSicialAuthority be testC100ValidationErrorWithSocialAuthority

File: src/functionalTest/resources/responses/bulk-scan-c100-transform-output.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          }
+        }
+      ],
+      "childLiveWith": "Applicant"
+    }
+  },
+  "warnings": [
+
+  ]
+}
```
Comment (hmcts-kodatala) [inline src/functionalTest/resources/responses/bulk-scan-c100-transform-output.json:66]: Scan document object not added here

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100Service.java [hunk 3]
```diff
@@ -41,9 +61,45 @@ public FormType getCaseType() {
...
+        if (!manualValidationErrors.isEmpty()) {
+            response.setStatus(Status.ERRORS);
+            response.getErrors().items.addAll(manualValidationErrors);
+        }
+
+        return response;
+    }
+
+    List<String> doManualValidation(Map<String, String> inputFieldsMap) {
+        List<String> errors = new ArrayList<>();
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100Service.java:79]: Rename Method name proper actuval validation and  add java doc which have section name of form

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100Service.java [hunk 4]
```diff
@@ -59,12 +115,32 @@ public BulkScanTransformationResponse transform(BulkScanTransformationRequest bu
...
         Map<String, Object> populatedMap = (Map<String, Object>) BulkScanTransformHelper
-            .transformToCaseData(transformConfigManager.getTransformationConfig(FormType.C100)
-                    .getCaseDataFields(), inputFieldsMap);
+            .transformToCaseData(new HashMap<>(transformConfigManager.getTransformationConfig(FormType.C100)
+                    .getCaseDataFields()), inputFieldsMap);
+
+        populatedMap.put(CHILD_LIVE_WITH_KEY, getChildLiveWith(inputFieldsMap));
+
+        Map<String, String> caseTypeAndEventId =
+            transformConfigManager.getTransformationConfig(FormType.C100).getCaseFields();
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100Service.java:124]: pls check scan documents url not managed this logic and add logic from other services

**Review by hmcts-kodatala — COMMENTED** [review]
added few cooments.

File: src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100ServiceTest.java (hunk 1 above)
Comment (saravanakumar-palanichamy) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100ServiceTest.java:202]: done.

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/controllers/BulkScanC100EndpointTest.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+            .body(bulkScanTransformRequest)
+            .when()
+            .contentType("application/json")
+            .post("/transform-exception-record");
+
+        response.then().assertThat().statusCode(HttpStatus.OK.value());
+
+        JSONAssert.assertEquals(bulkScanTransformResponse, response.getBody().asString(), true);
+    }
+}
```
[the remark this answers is not in the document]
Comment (saravanakumar-palanichamy) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/controllers/BulkScanC100EndpointTest.java:91]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100Service.java (hunk 4 above)
Comment (saravanakumar-palanichamy) [reply src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanC100Service.java:135]: yes based on other service only it was done

**Review by hmcts-kodatala — APPROVED** [review]
Ok

**Review by yogendraU — APPROVED** [review]

**Pull request merged**